runtime/compiler/z/codegen/J9BCDTreeEvaluator.cpp

```diff
@@ -6454,7 +6474,12 @@ J9::Z::TreeEvaluator::pdArithmeticVectorEvaluatorHelper(TR::Node * node, TR::Ins
          break;
       }
 
-   TR_ASSERT((immediateValue >> 8) == 0, "Decimal precision or adjustment value exceeds 1 byte");
+   TR_ASSERT_FATAL((immediateValue >> 8) == 0, "Decimal precision or adjustment value (%d) exceeds 1 byte", immediateValue);
```

Comment:
Will `immediateValue` (an `int32_t`) always be positive?  The multiplication by -1 on L6470 seems to suggest otherwise which would mean this assert would fail for negative values.  I'm not saying there's a problem here, just asking...

Reply:
I'll have to delegate this one. @NigelYiboYu could you please address @0xdaryl's question?

Reply:
Ping @NigelYiboYu

Reply:
Sorry for replying late. 
The `immediateValue` is always positive for now because the multiplication by -1 is never reached (dead code). I should remove both `VSDP` and `VMSP` cases later. They are just two unused Z instructions.

Reply:
Opened issue #5277.